src/Swift.Runtime/src/Metadata/ITypeMetadataCache.cs

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Diagnostics.CodeAnalysis;
+
+namespace Swift.Runtime;
+
+/// <summary>
+/// Represents an interface that defines how a cache for TypeMetadata should operate.
+/// The implementation of this interface needs to be thread safe.
+/// </summary>
+public interface ITypeMetadataCache
+{
+    /// <summary>
+    /// Returns true if and only if the cache contains an entry for t
+    /// </summary>
+    /// <param name="t">The type to look up in the cache</param>
+    /// <returns>true if and only if the cache contains t, false otherwise</returns>
+    bool Contains(Type t);
```

Comment:
I think the guidelines says to not abbreviate (xml docs would need to be fixed accordingly as well):

```suggestion
    bool Contains(Type type);
```